react/features/recent-list/components/RecentList.native.js

```diff
@@ -1,201 +1,241 @@
-                this._getColorForServerName(recentListEntry.serverName));
-        }
+    _onPress(url) {
+        const { dispatch } = this.props;
 
-        return avatarStyles;
+        dispatch(appNavigate(url));
     }
 
+    _toDisplayableItem: Object => Object
+
     /**
-     * Returns a style (color) based on the server name, so then the same server
-     * will always be rendered with the same avatar color.
+     * Creates a displayable list item of a recent list entry.
      *
-     * @param {string} serverName - The recent list entry being rendered.
      * @private
+     * @param {Object} item - The recent list entry.
      * @returns {Object}
      */
-    _getColorForServerName(serverName) {
-        let nameHash = 0;
-
-        for (let i = 0; i < serverName.length; i++) {
-            nameHash += serverName.codePointAt(i);
-        }
-
-        return styles[`avatarRemoteServer${(nameHash % 5) + 1}`];
+    _toDisplayableItem(item) {
+        const { _homeServer } = this.props;
+        const location = parseURIString(item.conference);
+        const baseURL = `${location.protocol}//${location.host}`;
+        const serverName = baseURL === _homeServer ? null : location.host;
+
+        return {
+            colorBase: serverName,
+            key: `key-${item.conference}-${item.date}`,
+            lines: [
+                this._toDateString(item.date),
+                this._toDurationString(item.conferenceDuration),
+                serverName
+            ],
+            title: location.room,
+            url: item.conference
+        };
     }
 
+    _toDisplayableList: () => Array<Object>
+
     /**
-     * Renders the conference duration if available.
+     * Transforms the history list to a displayable list
+     * with sections.
      *
-     * @param {Object} recentListEntry - The recent list entry being rendered.
      * @private
-     * @returns {ReactElement}
+     * @returns {Array<Object>}
      */
-    _renderConfDuration({ conferenceDurationString }) {
-        if (conferenceDurationString) {
-            return (
-                <View style = { styles.infoWithIcon } >
-                    <Icon
-                        name = 'timer'
-                        style = { styles.inlineIcon } />
-                    <Text style = { styles.confLength }>
-                        { conferenceDurationString }
-                    </Text>
-                </View>
-            );
+    _toDisplayableList() {
+        const { _recentList, t } = this.props;
+        const todaySection = NavigateSectionList.createSection(
+            t('recentList.today'),
+            'today'
+        );
+        const yesterdaySection = NavigateSectionList.createSection(
+            t('recentList.yesterday'),
+            'yesterday'
+        );
+        const earlierSection = NavigateSectionList.createSection(
+            t('recentList.earlier'),
+            'earlier'
+        );
+        const today = new Date().toDateString();
```

Comment:
I'm wondering why not use moment/dateUtils for all date logic?

Reply:
This is not a date formatting (although it looks like one), this is just a convenient way to trim the date to a date-only value (dropping the time portion) so then it's easy to compare with other date-only values (e.g. to decide if two dates are the same day or not)